test/experiment_tests.js

```diff
@@ -25,4 +24,30 @@ describe('Selector component', () => {
 			</Selector>)
 		assert.equal(selector.contains(<span>Blue</span>), true)
 	})
+
+	it ('Should return another random variant if the variant from the store is not in component', () => {
+		const initialState = {experiments: { button: 'yellow' }}
+		const mockStore = configureStore()
+		let store = mockStore(initialState)
+		shallow(<Experiment name="button" store={store}>
+				<Variant name="blue"><span>Blue</span></Variant>
+				<Variant name="red"><span>Red</span></Variant>
+				<Variant name="green"><span>Green</span></Variant>
+			</Experiment>).dive()
+		assert.equal(store.getActions().length, 1)
+		assert.equal(store.getActions()[0].type, 'RESET_EXPERIMENT_VARIANT')
+	})
+
+	it ('Should add experiment in store if it is not', () => {
+		const initialState = {experiments: { test: 'yellow' }}
+		const mockStore = configureStore()
+		let store = mockStore(initialState)
+		shallow(<Experiment name="button" store={store}>
+				<Variant name="blue"><span>Blue</span></Variant>
+				<Variant name="red"><span>Red</span></Variant>
+				<Variant name="green"><span>Green</span></Variant>
+			</Experiment>).dive()
+		assert.equal(store.getActions().length, 1)
+		assert.equal(store.getActions()[0].type, 'RESET_EXPERIMENT_VARIANT')
```

Comment:
what does this tests ?

Reply:
Same here, if the Experiment name is not in store, resetExperimentVariant will be called

Reply:
again, based on the test name it should do something else.